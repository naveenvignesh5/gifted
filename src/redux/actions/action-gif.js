import axios from "axios";

import { ACTION_TYPES } from "../actionTypes";
import { GIF_COUNT_PER_PAGE, API_ENDPOINT } from "../../constants";

const requestGif = () => ({
  type: ACTION_TYPES.REQUEST_FETCH_GIFS
});

const requestGifSuccess = payload => ({
  type: ACTION_TYPES.REQUEST_FETCH_GIFS_SUCCESS,
  payload
});

const requestGifFailure = error => ({
  type: ACTION_TYPES.REQUEST_FETCH_GIFS_FAILURE,
  error
});

const requestPlayToggle = (play = false) => ({
  type: play ? ACTION_TYPES.REQUEST_PLAY : ACTION_TYPES.REQUEST_PAUSE
});

const fetchGifs = searchObj => async dispatch => {
  try {
    dispatch(requestGif());
    const queryString = Object.keys(searchObj)
      .map(key => key + "=" + searchObj[key])
      .join("&");
    const res = await axios.get(
      `${API_ENDPOINT}/gifs/search?api_key=${process.env.REACT_APP_GIPHY_KEY}&${queryString}&limit=${GIF_COUNT_PER_PAGE}`
    ); // eslint-disable-line
    console.log(JSON.stringify(res.data));
    if (res.data) {
      dispatch(requestGifSuccess(res.data));
      return;
    }
    dispatch(requestGifFailure("no-data-found"));
  } catch (err) {
    console.log(err);
    dispatch(requestGifFailure(err));
  }
};

const togglePlay = play => dispatch => {
    dispatch(requestPlayToggle(play));
};

export { fetchGifs, requestGif, togglePlay };
