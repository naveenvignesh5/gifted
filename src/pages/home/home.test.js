import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

// page
import Home from "./home";

// actions
import { fetchGifs } from "../../redux/actions/action-gif";

const mockStore = configureStore([]);

describe("Home", () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      gif: {
        isFetching: false,
        isError: false,
        gifs: [],
        totalPages: 0,
        error: null
      }
    });

    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <Home />
      </Provider>
    );
  });

  it("snapshot renders with given state in redux store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

//   it("snapshot renders when dispatch fetchGifs", () => {
//     renderer.act(() => {
//         console.log(component.root.children);
//         component.root.props.fetchGifs({ q: 'cats', offset: 1 });
//     });

//     expect(store.dispatch).toHaveBeenCalledTimes(1);
//     expect(store.dispatch).toHaveBeenCalledWith(
//         fetchGifs({ q: 'cats', offset: 1 })
//     );
//   });
});
