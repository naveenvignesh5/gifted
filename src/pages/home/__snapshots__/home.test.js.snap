// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot renders with given state in redux store 1`] = `
<div
  className="container"
>
  <div
    className="search-bar-wrapper"
  >
    <input
      className="search-bar-input"
      defaultValue="cats"
      onChange={[Function]}
      placeholder="Search a gif..."
    />
    <button
      className="search-button"
      onClick={[Function]}
    >
      Search
    </button>
  </div>
  <div
    className="is-hidden-mobile"
  >
    <div
      className="paginatin-tab"
    >
      <div
        className="page-buttons-wrapper"
      >
        <button
          className="page-button"
          onClick={[Function]}
        >
          Prev
        </button>
        <button
          className="page-button is-active"
          onClick={[Function]}
        >
          1
        </button>
        <button
          className="page-button "
          onClick={[Function]}
        >
          2
        </button>
        <button
          className="page-button "
          onClick={[Function]}
        >
          3
        </button>
        <button
          className="page-button "
          onClick={[Function]}
        >
          4
        </button>
        <button
          className="page-button "
          onClick={[Function]}
        >
          5
        </button>
        <button
          className="page-button "
          onClick={[Function]}
        >
          6
        </button>
        <button
          className="page-button"
          onClick={[Function]}
        >
          Next
        </button>
      </div>
    </div>
  </div>
  <div
    className="gif-status-section"
  >
    <p
      className="status-text"
    >
      No Gifs Found.
    </p>
    ;
  </div>
</div>
`;
